Sources/TuistEnvKit/Commands/CommandRegistry.swift

```diff
@@ -46,6 +53,12 @@ public final class CommandRegistry {
             if processArguments().dropFirst().first == "--help-env" {
                 parser.printUsage(on: stdoutStream)
             } else if let parsedArguments = try parse() {
+                
+                let verbose = parsedArguments.get(verboseArgument) ?? false
+                
+                System.shared.verbose = verbose
+                FileHandler.shared.verbose = verbose
+            
```

Comment:
Ah it seems `--verbose` in tuistenv is a little trickier, as not all code paths go through here :/ From my testing it seems adding this confuses the parsers a little

```sh
$ tuistenv install 1.1.0
Warning: The release 1.1.0 couldn't be obtained from GitHub
Pulling source code
...
```

```sh
$ tuistenv --verbose install 1.1.0
Error: expected arguments: edit, build, generate, dump, create-issue, version, up, cache, focus, graph, init
```

We can omit it from tuistenv and do that separately if we can figure out how best to do it - my bad thought this would be a quick win 😞

Reply:
The logic which parses the command is what needed to change, it passed it through to the `tuist` commands OK but was failing on the specific route through `tuistenv`.

We need to be able to determine what the command name is over the argument , this is easy for `tuistenv`since we can ask if the parser can decode whats available in the arguments. 

Here's an example of what we could do in this instance:

```swift
struct ProcessArguments {
    let arguments: [String]
}

extension ProcessArguments {
    
    func commandName() -> String? {
        arguments.dropFirst().first(where: isCommand)
    }
    
    func isCommand(_ argument: String) -> Bool {
        argument.hasPrefix("-") == false
    }
    
}
```

One annoyance with this approach is older versions of `tuist` which do not support `--verbose` will fail saying `--verbose` is not an argument, but maybe it's not an issue because you will only use it knowing it's available.